core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/browsebars/BrowseBarComplete.java

```diff
@@ -85,141 +86,167 @@ private String printBreadCrumb() {
     if (componentInst != null || StringUtil.isDefined(getSpaceId())) {
-      List<SpaceInstLight> spaces = Collections.emptyList();
-
-      if (componentInst != null && !componentInst.isPersonal()) {
-        spaces = organizationController.getPathToComponent(getComponentId());
-      } else if (componentInst == null) {
-        spaces = organizationController.getPathToSpace(getSpaceId());
-      }
-      boolean firstSpace = true;
-      for (SpaceInstLight spaceInst : spaces) {
-        String spaceId = spaceInst.getId();
-        String href = "javascript:" + getSpaceJavascriptCallback() + "('" + spaceId + "')";
-        if (!isClickable()) {
-          href = "#";
-        }
-
-        if (!firstSpace) {
-          result.append(CONNECTOR);
-        }
-        result.append("<a href=\"").append(href).append("\"");
-        result.append(" class=\"space\"");
-        result.append(" id=\"space").append(spaceId).append("\"");
-        result.append(">");
-        result.append(Encode.forHtml(spaceInst.getName(language)));
-        result.append("</a>");
-
-        firstSpace = false;
-        emptyBreadCrumb = false;
-      }
-
-      if (componentInst != null) {
-        // Display component's label
-        if (!componentInst.isPersonal()) {
-          result.append(CONNECTOR);
-        }
-        result.append("<a href=\"");
-        if (!isClickable()) {
-          result.append("#");
-        } else if (StringUtil.isDefined(getComponentJavascriptCallback())) {
-          result.append("javascript:").append(getComponentJavascriptCallback()).append("('")
-              .append(getComponentId()).append("')");
-        } else {
-          result.append(URLUtil.getApplicationURL()).append(URLUtil.getURL(getSpaceId(),
-              getComponentId()));
-          if (ignoreComponentLink()) {
-            result.append("Main");
-          } else {
-            result.append(getComponentLink());
-          }
-        }
-        result.append("\"");
-        result.append(" class=\"component\"");
-        result.append(" id=\"bc_").append(componentInst.getId()).append("\"");
-        result.append(">");
-        result.append(Encode.forHtml(componentInst.getLabel(language)));
-        result.append("</a>");
-        emptyBreadCrumb = false;
-      }
+      breadcrumb.append(getSpacePath(componentInst));
+      breadcrumb.append(getComponent(componentInst));
     } else {
       if (getDomainName() != null) {
-        result.append(getDomainName());
-        emptyBreadCrumb = false;
+        breadcrumb.append(getDomainName());
       }
       if (getComponentName() != null) {
         if (getDomainName() != null) {
-          result.append(CONNECTOR);
+          breadcrumb.append(CONNECTOR);
         }
         if (getComponentLink() != null) {
-          result.append("<a href=\"").append(getComponentLink()).append("\">").append(
+          breadcrumb.append("<a href=\"").append(getComponentLink()).append("\">").append(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "<a href=\"" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)